web/src/main/java/com/epam/esm/security/filter/OktaJwtAuthenticationFilter.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class OktaJwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final Logger log = LoggerFactory.getLogger(OktaJwtAuthenticationFilter.class);
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final String TOKEN_TYPE_HINT_KEY = "token_type_hint";
+    private static final String ACCESS_TOKEN = "access_token";
+    private static final String TOKEN_KEY = "token";
+
+    private final RestTemplate template;
+    private final UserService userService;
+    private final UserAuthenticationService userAuthenticationService;
+    private final String introspectUrl;
+    private final HttpHeaders headers;
+
+    public OktaJwtAuthenticationFilter(UserService userService,
+                                       UserAuthenticationService userAuthenticationService,
+                                       @Value("${okta.introspect.url}") String introspectUrl,
+                                       @Value("${okta.client.id}") String clientId,
+                                       @Value("${okta.client.secret}") String clientSecret) {
+        this.userService = userService;
+        this.introspectUrl = introspectUrl;
+        this.userAuthenticationService = userAuthenticationService;
+        this.template = new RestTemplate();
+        headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        headers.setBasicAuth(clientId, clientSecret);
+    }
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain) throws ServletException, IOException {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication instanceof AnonymousAuthenticationToken) {
+            extractAuthBearerHeader(request)
+                    .map(this::parseJwt)
+                    .ifPresent(user -> authenticate(user, request));
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private Optional<String> extractAuthBearerHeader(HttpServletRequest request) {
+        String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
```

Comment:
How it it possible you already have access token here? I Have not found any place where you login a user using okta.